bokeh/themes/theme.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import, print_function
+
+import yaml
+
+from ..properties import HasProps
+
+# whenever we cache that there's nothing themed for a class, we
+# use this same dict instance, so we don't have a zillion empty
+# dicts in our caches.
+_empty_dict = dict()
+
+class Theme(object):
+    def __init__(self, filename=None, json=None):
+        if filename is not None:
+            if json is not None:
+                raise ValueError("Theme should be constructed from a file or from json not both")
+            f = open(filename)
+            try:
+                self._json = yaml.load(f)
+                # empty docs result in None rather than {}, fix it.
+                if self._json is None:
+                    self._json = {}
+            finally:
+                f.close()
+        elif json is not None:
+            self._json = json
```

Comment:
This could be a bit simpler/less nested:
```
if filename and json:
    raise ValueError("Theme should be constructed from a file or from json not both")
    
if filename:     
    with open(filename) as f:
        json = yaml.load(f) or dict()

self._json = json
```